pkg/cli/bicep/deployment_parameters.go

```diff
@@ -19,32 +19,23 @@ package bicep
 import (
 	"encoding/json"
 	"fmt"
-	"io/fs"
-	"os"
 	"strings"
 
+	"github.com/spf13/afero"
+
 	"github.com/radius-project/radius/pkg/cli/clients"
 )
 
 // ParameterParser is used to parse the parameters as part of the `rad deploy` command. See the docs for `rad deploy` for examples
 // of what we need to support here.
 type ParameterParser struct {
-	FileSystem fs.FS
-}
-
-type OSFileSystem struct {
+	FileSystem afero.Fs
```

Comment:
instead of writing a filesystem abstraction I added this `afero` library. Seems to be widely used, I replaced the `OSFileSystem` with it. LMK if I should revert

Reply:
are there added functionalities that we get using `afero` or is the main benefit that's it's an abstraction layer over `os`?

Reply:
Good find! I added it here because the stdlib `fs` does not include a filesystem abstraction with `CreateFile`. The benefit of adding `afero` was to not have to implement and maintain this ourselves. Since we already indirectly depend on this library, and that it's a pretty widespread dependency, I'd say we should be okay to use it. Thoughts? @brooke-hamilton

Reply:
@rynowak thoughts on this?

Reply:
What's the savings of using this library? I don't have fundamental concerns about it, I just can't tell from the diff what's going to change.

Reply:
I'd also really like to remove viper as a dependency. We've got a lot of workarounds in place for for behaviors that aren't helpful to us.

Reply:
I removed `afero` as a dependency and wrote a simple filesystem abstraction that we can use